keytree/keytree.md

```diff
@@ -0,0 +1,135 @@
+# Key tree
+
+This is a _key blinding scheme_ for deriving hierarchies of public keys.
+
+The most important feature of this scheme is that a set of _public_ keys can be derived from a _public_ key,
+without the use of private keys. This allows a piece of software generate unique receiving addresses
+without having the private key material available. E.g. an online merchant may keep only public keys on the server and generate invoices with unique keys without compromising security of the private keys.
+
+## Definitions
+
+### Scalar
+
+A _scalar_ is an integer modulo [Ristretto group](https://ristretto.group) order `|G| = 2^252 + 27742317777372353535851937790883648493`.
+
+Scalars are encoded as 32-byte strings using little-endian convention.
+
+Every scalar is required to be in a canonical (reduced) form.
+
+### Point
+
+A _point_ is an element in the [Ristretto group](https://ristretto.group).
+
+Points are encoded as 32-byte [data types](#data-type) in _compressed Ristretto form_.
+
+
+### Base point
+
+Ristretto base point in compressed form:
+
+```
+B = e2f2ae0a6abc4e71a884a961c500515f58e30b6aa582dd8db6a65945e08d2d76
+```
+
+### Derivation key
+
+Uniformely random secret 32-byte string used to derive blinding factors.
+
+### Xprv
+
+Stands for _extended private key_: consists of a secret [scalar](#scalar) and a [derivation key](#derivation-key).
+
+```rust
+struct Xprv {
+  scalar: Scalar,
+  dk: [u8; 32]
```

Comment:
Wait, what does the notation on this line mean?

Reply:
i lazily used Rust syntax for fixed-length arrays: `[T, N]`: N items of type T. `u8` is 8-bit unsigned integer aka "byte".

Reply:
`[T]` is a type for "slice" that's of a dynamically known size, while `[T;N]` is an "array" of statically-known size.

Reply:
Oh, nice, thanks. I just got to the arrays section of this ole Rust book